enhancements/monitoring/replace-prometheus-adapter-with-metrics-server.md

```diff
@@ -0,0 +1,294 @@
+  metrics-server settings. 
+
+  ```go
+  type ClusterMonitoringConfiguration struct {
+    ...
+    // `MetricsServerConfig` defines settings for the MetricsServer component.
+    MetricsServerConfig *MetricsServerConfig `json:"metricsServer,omitempty"`
+  }
+  ```
+
+  When `enabled` is set `true`, the cluster monitoring operator configures the Metrics Server in `openshift-monitoring` namespace
+
+  ```go
+  // The `MetricsServerConfig` resource defines settings for the MetricsServer component.
+  type MetricsServerConfig struct {
+    // `Enabled` is a Boolean flag that enables or disables metrics-server.
+    Enabled *bool `json:"enabled,omitempty"`
+    // Defines the nodes on which the pods are scheduled.
+    NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+    // Defines tolerations for the pods.
+    Tolerations []v1.Toleration `json:"tolerations,omitempty"`
+    // Defines resource requests and limits for the Prometheus container.
+    Resources *v1.ResourceRequirements `json:"resources,omitempty"`
+  }
+  ```
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Configuration/Enablement
+
+Metrics Server is not enabled by default. It must be explicitly enabled by the user. It is enabled by setting 
+`enabled: true` in the ConfigMap `openshift-monitoring/cluster-monitoring-config` under `metricsServer` section
+
+#### Resource Impact
+
+Once metrics-server is enabled, by default 2 replicas of `metrics-server` instances are being deployed in the openshift-monitoring namespace.
+
+As of OpenShift 4.13 the following idle resource usage have been measured:
+
+Kubelet: TBD
+Metrics Server: TBD
+
+**TODO:** Capture the details on performance testing done from [MON-3210](https://issues.redhat.com/browse/MON-3210)
+
+
+### Risks and Mitigations
+
+One of the risk is resource impact on larger clusters. We need to do performance testing on larger OpenShift cluster to measure the impact on Kubelet when metrics-server is deployed.
+
+[MON-3210](https://issues.redhat.com/browse/MON-3210) is created for doing this.
+
+### Drawbacks
+
+N/A
+
+
+## Design Details
+
+### Open Questions [optional]
+
+[Custom Metrics Autoscaler](https://github.com/openshift/custom-metrics-autoscaler-operator)(CMA) also deploys [metrics-server](https://github.com/openshift/custom-metrics-autoscaler-operator/blob/39aa2b9d5a0db2206d4963fe2e551178e7d369ad/controllers/keda/kedacontroller_controller.go#L204) when user creates keda controller after its installed, will this affect Metrics Server we deploy in monitoring stack?
+
+### Test Plan
+
+The proposed solution will be tested with two suites:
+
+* Unit tests - running independently of each other and without side effects
+* End to end tests - in the same repository as the cluster-monitoring-operator. These tests will be executed against an actual Kubernetes cluster and are intended to verify end-to-end functionality of the entire system.
```

Comment:
With more details: there should be an end-to-end test that
* Enables metrics-server
* Verifies that the resource metrics API is working
* Disables metrics-server
* Verifies that the resource metrics API is working
During all the test, it will ensure that there's no disruption of the resource metrics API.